cmake/GetVulkan.cmake

```diff
@@ -0,0 +1,79 @@
+        endif()
+    endif()
+
+    # now look for glslang stuff
+    if(Vulkan_FOUND)
+        unset(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE CACHE)
+        unset(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE)
+	find_package(glslang REQUIRED)
+        if(glslang_FOUND)
+            find_program(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE glslangValidator HINTS ${CMAKE_INSTALL_PREFIX}/bin)
+        else()
+	    message(STATUS "glslang NOT FOUND!!!")
+            set(Vulkan_FOUND FALSE CACHE INTERNAL "" FORCE)
+            set(Vulkan_FOUND FALSE)
+            clearVars()
+        endif()
+    endif()
+
+    if(Vulkan_FOUND)
+        # message(STATUS "  Found Vulkan, version ${Vulkan_VERSION} ${Vulkan_LIBRARIES} ${Vulkan_INCLUDE_DIRS}")
+        # Since we now have both Vulkan and glslang found, they will use the system wide Vulkan
+        # set(Vulkan_DEP_STR Vulkan_ext)
+        # set(Vulkan_EXTERNAL Yes)
+        set(Vulkan_INCLUDE_DIRS ${CMAKE_INSTALL_PREFIX}/include)
+        # set(Vulkan_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/_deps/vulkan/src/Vulkan_ext/Vulkan-Headers/include)
+        set(Vulkan_LIBRARIES ${CMAKE_INSTALL_PREFIX}/lib/libvulkan.so)
+        # set(Vulkan_LIBRARIES /usr/lib/x86_64-linux-gnu/libvulkan.so)
+        set(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE ${CMAKE_INSTALL_PREFIX}/bin/glslangValidator)
+        # set(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE /usr/bin/glslangValidator)
+        message(STATUS "  Found Vulkan, version ${Vulkan_VERSION} ${Vulkan_LIBRARIES} ${Vulkan_INCLUDE_DIRS}")
+
+        set(Vulkan_VERSION "${Vulkan_VERSION}" CACHE INTERNAL "" FORCE)
+    else()
+        message(STATUS "Building VULKAN from source")
+        EXTERNALPROJECT_ADD(Vulkan_ext
+                            GIT_REPOSITORY https://github.com/ILLIXR/ILLIXR-vulkan.git
+                            GIT_TAG ca2e256f2ef5bcdd0ffd05f7712bd03897d9c991
+                            PREFIX ${CMAKE_BINARY_DIR}/_deps/vulkan
+                            CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX} -DCMAKE_BUILD_TYPE=Release -DBUILD_PARALLEL_LEVEL=${BUILD_PARALLEL_LEVEL}
+                            BUILD_COMMAND cmake --build . --parallel ${BUILD_PARALLEL_LEVEL}
+                            INSTALL_COMMAND ""
+                            )
+        set(Vulkan_DEP_STR Vulkan_ext)
+        set(Vulkan_EXTERNAL Yes)
+        set(Vulkan_INCLUDE_DIRS ${CMAKE_INSTALL_PREFIX}/include)
+        # set(Vulkan_INCLUDE_DIRS ${CMAKE_BINARY_DIR}/_deps/vulkan/src/Vulkan_ext/Vulkan-Headers/include)
+        set(Vulkan_LIBRARIES ${CMAKE_INSTALL_PREFIX}/lib/libvulkan.so)
+        # set(Vulkan_LIBRARIES /usr/lib/x86_64-linux-gnu/libvulkan.so)
+        set(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE ${CMAKE_INSTALL_PREFIX}/bin/glslangValidator)
+        # set(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE /usr/bin/glslangValidator)
+    endif()
+endif()
+
+set(Vulkan_GLSLANG_VALIDATOR_EXECUTABLE ${Vulkan_GLSLANG_VALIDATOR_EXECUTABLE} CACHE INTERNAL "" FORCE)
```

Comment:
why are we emptying this variable? I think this is the cause of   timewarp_vk not finding glslangValidator.